src/main/java/edu/byu/cs/autograder/score/Scorer.java

```diff
@@ -130,18 +131,21 @@ private Submission successfullyProcessSubmission(Rubric rubric, Rubric lateAppli
      * Note that this operation will be performed carefully so that existing RubricItem's in Canvas
      * will not be overwritten by the operation.
      *
-     * @param rubric A {@link Rubric} containing values to set in Canvas.
-     *               Any items not set will be populated with their value from Canvas.
-     * @param phase The phase being graded
-     * @param netId Net ID of student being
+     * @param submission Submission object of approved submission
      * @param penaltyPct The approved GIT_COMMITS penalty percentage
      * @throws GradingException When preconditions are not met.
      * @throws DataAccessException When the database cannot be reached.
      */
-    public static void attemptSendToCanvas(Rubric rubric, Phase phase, String netId, int penaltyPct, String commitPenaltyMsg) throws GradingException, DataAccessException {
+    public static void attemptSendApprovedScoreToCanvas(Submission submission, int penaltyPct, String commitPenaltyMsg) throws GradingException, DataAccessException {
```

Comment:
I don't support this change in the call signature of this method.

The previous call signature has been intentionally destructured to require only the lowest possible  set of objects, and data. The method is clearly documented as "sending a provided rubric to Canvas," and it's already doing exactly that job.

Passing in more complicated data (`Submission`) and adding more dependencies (`new LateDayCalculator()`) is contradictory to solid software architecture and is violating several important principles. As already documented in the call signature, any/all penalties (late, commit verification, or otherwise) should already be applied before calling this method.

It's sad to see perceived urgency leading to unnecessary technical complexity and technical debt.

Reply:
Case in point, did you mean to also adjust the other call signature that used to have this name? It is also submitting Rubrics to Canvas, but it doesn't have this changed behavior. What new issues will that cause? Are there other use-cases that will still be left open because they are leveraging a different call signature?

How do we know that the late-penalty will not be applied twice for students submitting regularly (Once by the main system, and a second time by this system)?